internal/cli/atlas_alertconfig_create.go

```diff
@@ -0,0 +1,221 @@
+	store                           store.AlertConfigurationCreator
+}
+
+func (opts *atlasAlertConfigCreateOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *atlasAlertConfigCreateOpts) Run() error {
+	alert := opts.buildAlertConfiguration()
+	result, err := opts.store.CreateAlertConfiguration(alert)
+
+	if err != nil {
+		return err
+	}
+
+	return json.PrettyPrint(result)
+}
+
+func (opts *atlasAlertConfigCreateOpts) buildAlertConfiguration() *atlas.AlertConfiguration {
+
+	alert := new(atlas.AlertConfiguration)
+
+	alert.GroupID = opts.ProjectID()
+	alert.EventTypeName = strings.ToUpper(opts.event)
+	alert.Enabled = &opts.enabled
+
+	if opts.matcherFieldName != "" {
+		match := new(atlas.Matcher)
+		match.FieldName = strings.ToUpper(opts.matcherFieldName)
+		match.Operator = strings.ToUpper(opts.matcherOperator)
+		match.Value = strings.ToUpper(opts.matcherValue)
+		alert.Matchers = []atlas.Matcher{*match}
+	}
+
+	if opts.metricThresholdMetricName != "" {
+		metric := new(atlas.MetricThreshold)
+		metric.MetricName = strings.ToUpper(opts.metricThresholdMetricName)
+		metric.Operator = strings.ToUpper(opts.metricThresholdOperator)
+		metric.Threshold = opts.metricThresholdThreshold
+		metric.Units = strings.ToUpper(opts.metricThresholdUnits)
+		metric.Mode = strings.ToUpper(opts.metricThresholdMode)
+		alert.MetricThreshold = metric
+	}
+
+	notification := new(atlas.Notification)
+	notification.TypeName = strings.ToUpper(opts.notificationTypeName)
+	notification.DelayMin = &opts.notificationDelayMin
+	notification.IntervalMin = opts.notificationIntervalMin
+	notification.TeamID = opts.notificationTeamID
+	notification.Username = opts.notificationUsername
+
+	switch notification.TypeName {
+
+	case victor:
+		notification.VictorOpsAPIKey = opts.apiKey
+		notification.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+
+	case slack:
+		notification.VictorOpsAPIKey = opts.apiKey
+		notification.VictorOpsRoutingKey = opts.notificationVictorOpsRoutingKey
+
+	case datadog:
+		notification.DatadogAPIKey = opts.apiKey
+		notification.DatadogRegion = strings.ToUpper(opts.notificationRegion)
+
+	case email:
+		notification.EmailAddress = opts.notificationEmailAddress
+
+	case flowdock:
+		notification.FlowdockAPIToken = opts.token
+		notification.FlowName = opts.notificationFlowName
+		notification.OrgName = opts.notificationOrgName
+
+	case sms:
+		notification.MobileNumber = opts.notificationMobileNumber
+
+	case group, user, org:
+		notification.SMSEnabled = &opts.notificationSmsEnabled
+		notification.EmailEnabled = &opts.notificationEmailEnabled
+
+	case ops:
+		notification.OpsGenieAPIKey = opts.apiKey
+		notification.OpsGenieRegion = opts.notificationRegion
+
+	case pager:
+		notification.ServiceKey = opts.notificationServiceKey
+
+	}
+
+	alert.Notifications = []atlas.Notification{*notification}
```

Comment:
instead of doing `new(Notification)` you can do `:= Notification{}` and not use a pointer here
also please consider that the following is also valid/ok
```go
notification := atlas.Notification{
TypeName: strings.ToUpper(opts.notificationTypeName),
DelayMin: &opts.notificationDelayMin,
IntervalMin: opts.notificationIntervalMin,
TeamID: opts.notificationTeamID,
Username: opts.notificationUsername,
}
```